src/useMultiRootEditor.tsx

```diff
@@ -530,12 +530,9 @@ const useMultiRootEditor = ( props: MultiRootHookProps ): MultiRootHookReturns =
 				} );
 			};
 
-			const _updateEditorData = () => {
-				// If any of the roots content has changed, set the editor data.
-				// Unfortunately, we cannot set the editor data just for one root,
-				// so we need to overwrite all roots (`nextProps.data` is an
-				// object with data for each root).
-				instance.data.set( data, { suppressErrorInCollaboration: true } as any );
+			const _updateEditorData = ( roots: Array<string> ) => {
+				const dataToUpdate = roots.reduce( ( result, rootName ) => ( { ...result, [ rootName ]: data[ rootName ] } ), {} );
+				instance.data.set( dataToUpdate, { suppressErrorInCollaboration: true } as any );
```

Comment:
Consider using `Object.create( null )` as initial value of reduce.

Reply:
Good point, fixed in f09915b7593bdbb57cca4e04221b2088fb69e43f.